### DISCUSSION TITLE
Auth with custom fields

### DISCUSSION THREAD

**Discussion by rafaelcascalho** [adonisjs/core#1306] [Help] [answered]
Hey guys!! Hope everyone is safe. :bow: 

### Context
I have a defined database that my client very much insists on using, and on his database, the `uid` field is not used for `login` and I also require a third field. I need `web` and `api` guards.

### Question
How can I do that using the built-in `auth`? I mean, I'm thinking about using `attempt(fields)`, but seems that it requires that the first field be a uid field, which in this case it's not. So, I'm wondering about the possibilities.

**Comment by thetutlage** [comment, marked as answer]
Simply fetch the user yourself from the database, verify the password and then use `auth.login` method. This is how it should roughly look like.

```ts
import Hash from '@ioc:Adonis/Core/Hash'

const user = await User.query().where().firstOrFail()
const isValid = await Hash.verify(user.password, passwordSubmittedInForm)

if (!isValid) {
  // return error
}

await auth.login(user)
```
